internal/statistics.go

```diff
@@ -140,38 +139,30 @@ func (s *statistics) sendStatsToOwner(stats *bytes.Buffer) {
 	}
 }
 
-func (s *statistics) registerMetrics() {
+func (s *statistics) collectMetrics() {
+	s.collectOSMetrics()
+	s.collectRuntimeMetrics()
+}
+
+func (s *statistics) collectRuntimeMetrics() {
 	var m runtime.MemStats
 	runtime.ReadMemStats(&m)
+	s.registerGauge("runtime.availableProcessors", strconv.Itoa(runtime.NumCPU()))
```

Comment:
I do not think that you need to design with gauges, it was used in Java since the monitor also uses it but for non-java i think it is just extra overhead. you can just pass around the string buffer stream to be filled instead.